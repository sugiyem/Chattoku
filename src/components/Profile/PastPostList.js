import { SectionList } from "react-native";
import PastPostCard from "./PastPostCard";

const PastPostList = ({ postsData, Header }) => {
  const renderItem = ({ item }) => {
    return <PastPostCard {...item} />;
  };

  return (
    <SectionList
      sections={[{ data: postsData }]}
      renderItem={renderItem}
      renderSectionHeader={Header}
    />
  );
};

export default PastPostList;
