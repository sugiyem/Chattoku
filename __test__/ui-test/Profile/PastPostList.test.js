import React from "react";
import renderer from "react-test-renderer";
import PastPostList from "../../../src/components/Profile/PastPostList";

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: jest.fn()
  })
}));

describe("Test Past Post List UI", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <PastPostList
          postsData={[
            {
              forumData: { img: "forum-img", title: "First Forum" },
              postData: {
                timestamp: new Date(2022, 4, 8),
                lastEdited: new Date(2022, 5, 5),
                title: "Post Title",
                content: "Post Content"
              }
            }
          ]}
          Header={() => <></>}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
