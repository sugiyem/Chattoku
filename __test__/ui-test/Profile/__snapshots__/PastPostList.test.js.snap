// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Past Post List UI Renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "forumData": Object {
              "img": "forum-img",
              "title": "First Forum",
            },
            "postData": Object {
              "content": "Post Content",
              "lastEdited": 2022-06-04T17:00:00.000Z,
              "timestamp": 2022-05-07T17:00:00.000Z,
              "title": "Post Title",
            },
          },
        ],
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    />
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "whitesmoke",
              "borderColor": "black",
              "borderRadius": 10,
              "borderWidth": 1,
              "marginBottom": 10,
              "marginLeft": 10,
              "marginRight": 10,
              "marginTop": 10,
              "overflow": "hidden",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "turquoise",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "fontSize": 16,
                "justifyContent": "flex-start",
                "paddingBottom": 5,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 5,
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "forum-img",
              }
            }
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "borderColor": "black",
                  "borderRadius": 25,
                  "borderWidth": 1,
                  "height": 50,
                  "width": 50,
                },
              ]
            }
            testID="forumImg"
          />
          <Text
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "400",
                  "marginLeft": 10,
                },
              ]
            }
            testID="forumName"
          >
             
            First Forum
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 13,
                  "marginLeft": "auto",
                },
              ]
            }
            testID="postDate"
          >
             
            May 08 2022
             
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "alignSelf": "center",
                "fontSize": 18,
                "fontWeight": "500",
                "paddingBottom": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
                "paddingTop": 10,
              },
            ]
          }
          testID="postTitle"
        >
          Post Title
        </Text>
        <Text
          style={
            Array [
              Object {
                "maxHeight": 100,
                "overflow": "hidden",
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
          testID="postContent"
        >
           
          Post Content
           
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 13,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              },
            ]
          }
        >
          (Last Edited: 
          Jun 05 2022
          )
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "aquamarine",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 5,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              }
            }
            testID="goToPost"
          >
            <Text>
               See Full Post 
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "aquamarine",
                "borderRadius": 10,
                "borderWidth": 1,
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "marginBottom": 5,
                "marginLeft": 5,
                "marginRight": 5,
                "marginTop": 5,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              }
            }
            testID="goToForum"
          >
            <Text>
              Go To Forum
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    />
  </View>
</RCTScrollView>
`;
